### PR TITLE
Option to load all dock widgets

### PR DESCRIPTION
**Pull request by mstabrin**
# Description
Sometimes we want to load all dock widgets of a certain plugin, or even have dock widgets of multiple dock widgets on startup.
Therefore, I propose a new method to start all dock widgets and provide multiple plugins:

```
napari -w plugin1 dock1 dock2 -w plugin2 __all__
```

I chose the keyword `__all__` here, but it could also be the default behavior when no dock widget is provided!?

image

Since this might clutter the GUI, I also propse a new `-t` option that will lead to a tabified version:

```
napari -w plugin1 dock1 dock2 -w plugin2 __all__ -t
```
image

## Type of change
- [X] New feature (non-breaking change which adds functionality?

# How has this been tested?

### CODE REVIEW

**Review by tlambert03 — COMMENTED** [review]
thanks @mstabrin, great idea.  I'm for it.

a couple thoughts/questions first:  What do you (and others) think about making `__all__` just be the default?  Currently, if I recall correctly, if a plugin offers multiple widgets, but you don't specify one, you get an exception, is that right?  Maybe rather than erroring out we could just open them all (without requiring the explicit, but possibly harder to discover `__all__` flag)?  If someone doesn't like that behavior, they can then of course use the existing API to select a subset.  Thoughts?

Also, I do like the `tabify` argument on `add_plugin_dock_widget` 👍 ... but being ever wary of adding new public APIs, I wonder if there's any way to avoid adding it to our command line program.  I would be totally fine having the auto behavior of the cli being `tabify = len(widgets) > 1`.  Thoughts there?

**Comment by Czaki** [comment]
> recall correctly, if a plugin offers multiple widgets, but you don't specify one, you get an exception, is that right

in my case, it opens the first widget form list (in stable release)

**Comment by tlambert03** [comment]
> in my case, it opens the first widget form list (in stable release)

ahh ... ok, in that case we probably shouldn't change the behavior to opening them all... in which case I'm happy with `__all__` as a special flag.

**Comment by mstabrin** [comment]
Yes it does load the first dock widget in the list. I am also totally fine to always open multiple plugins as tabs, that was my first thought, but I did not want to break to much of the existing code :)

**Comment by mstabrin** [comment]
I added some documentation for the otherwise very hard to discover `__all__` option.

**Review by tlambert03 — APPROVED** [review]
I could be convinced either way on the `--tabify` cli argument, vs just auto-tabifying multiple widgets.  I'll approve now and leave it to you and anyone with a stronger opinion :)

File: napari/__main__.py
Comment (tlambert03) [inline napari/__main__.py]:
```suggestion
                        if name == 'dock' and pname == _pname:
                             wnames = _wnames
                             break
```

Comment (tlambert03) [reply napari/__main__.py]: it's also perhaps worth noting that this will work for npe2 plugins, but not for older plugins (which use [this fallback search here](https://github.com/napari/napari/blob/main/napari/_qt/qt_main_window.py#L679-L682)).  that's ok with me :joy: ... but just a heads up

Comment (mstabrin) [reply napari/__main__.py]: Hmm true, let me think about it :D

**Comment by mstabrin** [comment]
Ok non-npe2 plugins are confusing, for some reason they cannot be tabified, but I do not understand why.
Even just loading everything manually and drag+drop them on top of each other for the tabification makes the underlying plugin widget disappear.

Therefore, npe2 plugins are now loaded before non-npe2 plugins and npe2 plugins can be tabified while non-npe2 plugin will not be tabified and an information is printed.

Any ideas why this happens?

**Comment by alisterburt** [comment]
great suggestion and another great PR @mstabrin - fantastic work! I'm not sure why the behaviour is strange for original plugin engine plugins but I'm okay with ignoring it, there should be very few npe1 plugins around/being created moving forwards. 

For the tabify CLI arg, how about we don't add a CLI arg here, set (and document) the default behaviour. This default should probably be to auto tabify if more than one widget, I would expect developers to group widgets together themselves if they wanted one bigger widget. We can then have a separate discussion about tabification from the CLI

It would also be great to get this merged next week so it can be included in 0.4.17!

**Comment by mstabrin** [comment]
Ok, Tabify is now the default for npe2 plugins without a cli option :)

File: napari/__main__.py
Comment (alisterburt) [inline napari/__main__.py]:
```suggestion
```

**Comment by alisterburt** [comment]
Thanks again @mstabrin ! Will merge here when CI has finished

**Pull request merged**
